mixins/labelled-mixin.js

```diff
@@ -101,14 +111,18 @@ export const LabelledMixin = superclass => class extends superclass {
 
 			if (labelElem) {
 				this.label = getLabel(labelElem);
+				// TODO: how to validate empty label?
```

Comment:
The validation in this case wasn't handled before, and still isn't. The case is an element `labelled-by` another element that _exists_, but has an empty label.

The reason we can't validate this case right now is because `LabelMixin` implementations typically call `updateLabel()` inside their own `updated()` callback, which means there's an initial render where the thing hasn't set the label yet, and then it sets it and everything gets re-rendered. So if we validated right here, we'd falsely fail.

The only way I see to get around this is for `LabelMixin` implementations to synchronously call `updateLabel` in the `setter` for whatever is affecting the label -- and we're well aware of the perils in doing that. Alternately, we could delay the validation somehow? @dbatiste interested in your thoughts here.

Reply:
This is interesting from the editor's point of view too, I guess. The label syncs down to TinyMCE on init and updates if the label is changed afterwards, but if we can't bank on an explicit `set`, then we would sync an empty title down to TinyMCE and never change that. 🤔

Reply:
So in almost all scenarios -- even when `labelled-by` is used -- the label is there right away on first render and everything is peachy.

But in some weird scenarios (like one of the tests where it replaces the label element with a new one), the label isn't there on the first render. But even then, the `label` attribute _eventually_ gets set, which would trigger a re-render of the editor.

Reply:
Ah, okay, I misunderstood your first comment - we can't validate here because we can't differentiate between the case where the label is going to stay empty vs. it getting set a bit later in the lifecycle.

Reply:
Hence we can always bank on it being set later provided the consumer hasn't just gone and tossed a legitimately empty label in.

Reply:
Exactly yeah. One weird/maybe fragile option would be to just "wait a bit" and come back and if the label is still missing then we error.

Reply:
I am a little worried about the case where the label data is coming from an HM response which would be delayed.

Reply:
Is anything using `labelled-by` though yet? A [quick code search](https://search.d2l.dev/search?project=Brightspace&project=BrightspaceHypermediaComponents&project=BrightspaceUI&project=BrightspaceUILabs&project=lms&full=%22labelled-by%22&defs=&refs=&path=&hist=&type=&xrd=&nn=5) doesn't appear to show anything.

I'd really like to encourage teams to not render the list items (or inputs, or whatever) until they have all the data they need, and forcing this from the start before anyone does it in the first place seems like the right approach.

Reply:
I was poking around earlier and I don't think anything is technically using it at the moment, but there is [one spot that should be](https://search.d2l.dev/xref/BrightspaceHypermediaComponents/foundation-components/components/activity/editor/collection/d2l-activity-editor-collection-add.js?r=dd8f23fe#140).  I am going to put up a PR to add the `labelled-by` attribute here - the mixins are all set up, but I think the wire-up was forgotten.